client/fingerprint/env_aws.go

```diff
@@ -114,6 +115,17 @@ func (f *EnvAWSFingerprint) Fingerprint(cfg *config.Config, node *structs.Node)
 		"public-ipv4",
 		"placement/availability-zone",
 	}
+
+	// Keys that should be marked as unique
+	unique := map[string]struct{}{
```

Comment:
Make the above map a `map[string]bool` to not accidentally mess up new keys. Add more docs as to why.